internal/datastore/postgres/postgres.go

```diff
@@ -123,22 +124,45 @@ func NewPostgresDatastore(
 }
 
 func newPostgresDatastore(
-	url string,
+	pgURL string,
 	options ...Option,
 ) (datastore.Datastore, error) {
 	config, err := generateConfig(options)
 	if err != nil {
 		return nil, fmt.Errorf(errUnableToInstantiate, err)
 	}
 
+	// check for plan_cache_mode and, if not specified, set to `force_custom_plan`.
+	// See: https://spicedb.dev/d/force-custom-plan for more information.
+	parsed, err := pgxpool.ParseConfig(pgURL)
+	if err != nil {
+		return nil, fmt.Errorf(errUnableToInstantiate, err)
+	}
+
+	if existing, ok := parsed.ConnConfig.Config.RuntimeParams["plan_cache_mode"]; ok {
+		log.Info().Str("plan_cache_mode", existing).Msg("found plan_cache_mode in DB URL; leaving as-is")
+	} else {
+		parsedURL, err := url.Parse(pgURL)
+		if err == nil {
+			log.Warn().Msg("could not parse URL to automatically add `plan_cache_mode=force_custom_plans`; likely an old-style DSN")
+		}
+
+		queryParams := parsedURL.Query()
+		queryParams.Add("plan_cache_mode", "force_custom_plan")
+		parsedURL.RawQuery = queryParams.Encode()
+
+		pgURL = parsedURL.String()
+		log.Info().Msg("set plan_cache_mode='force_custom_plan' in Postgres DB URL")
```

Comment:
I think all of the logging related to this should include the spicedb.dev URL.
All spicedb.dev URLs should probably be stored under the same key (maybe "details_url"?).